### PR TITLE
Add Password Strength Checker to Registration

### PR DESCRIPTION
**Pull request by stevenelleman**

### CODE REVIEW

**Comment by stevenelleman** [comment]
Screenshot 2024-11-03 at 9 06 16 PM

Screenshot 2024-11-03 at 9 06 03 PM

**Comment by stevenelleman** [comment]
Added more explicit first warning and cleaned up a bit: 
image

File: apps/frontend/package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "eslint-config-next": "14.2.13",
     "postcss": "^8",
     "tailwindcss": "^3.4.1",
-    "typescript": "^5"
+    "typescript": "^5",
+    "zxcvbn": "^4.4.2"
```
Comment (AndrewCLu) [inline apps/frontend/package.json:52]: does it make sense to use typescript version of this instead? https://www.npmjs.com/package/@zxcvbn-ts/core

File: apps/frontend/src/pages/register.tsx [hunk 2]
```diff
@@ -166,6 +169,22 @@ const Register: React.FC = () => {
   };
 
   const handleRegisterWithPassword = async (password: string) => {
+
+    // Check password strength
+    const passwordCheck = zxcvbn(password);
+    if (passwordCheck && passwordCheck.score < 3) {
```
Comment (AndrewCLu) [inline apps/frontend/src/pages/register.tsx:175]: why 3? seems arbitrary, can you add a comment and link to some docs explaining this?

File: apps/frontend/src/pages/register.tsx [hunk 3]
```diff
@@ -166,6 +169,22 @@ const Register: React.FC = () => {
...
+      let errorMsg: string = "Weak password, submit new one.";
+      if (passwordCheck.feedback?.warning) {
+        errorMsg += ` ${passwordCheck.feedback.warning}.`;
+      }
+      if (passwordCheck.feedback.suggestions.length > 0) {
```
Comment (AndrewCLu) [inline apps/frontend/src/pages/register.tsx:180]: i feel like feedback suggestions for a password, esp displayed as a toast, doesn't send the right ux message. this is more of a master password than an auth password, so we should just have the user retry

**Comment by AndrewCLu** [comment]
> Screenshot 2024-11-03 at 9 06 16 PM Screenshot 2024-11-03 at 9 06 03 PM

This message is kinda jank. Can we instead just have a generic toast that says something like "use less common words, add numbers and symbols, etc."

File: apps/frontend/package.json (hunk 1 above)
Comment (stevenelleman) [reply apps/frontend/package.json:52]: Thanks for finding, looked and didn't find it originally.

**Comment by stevenelleman** [comment]
> This message is kinda jank. Can we instead just have a generic toast that says something like "use less common words, add numbers and symbols, etc."

I don't think you pulled the most recent change. I fixed the styling: https://github.com/cursive-team/connections/pull/165#issuecomment-2454202250

Updated to remove suggestion, so it's not that relevant anyways: 
image

**Review by AndrewCLu — APPROVED** [review]
Update toast text to "Weak password, try adding numbers, symbols, and using less common words."

**Pull request merged**